Introduction-to-Grails-Development-with-Intellij/index.md

```diff
@@ -0,0 +1,293 @@
+
+To run the application run the following command
+
+```bash
+grails run-app
+```
+The command will make the web application to run via a browser. You can type the highlighted command in the browser.
+
+![CREATE-APP](/engineering-education/how-to-replicate-postgresql-database/grails-run-app.png).
+
+![BROWSER VIEW](/engineering-education/how-to-replicate-postgresql-database/grails-first-web-app.png).
+
+### Folder structure and directories
+
+The grails application has the following folder structure.
+
+```bash
+├── .gradle
+├── build
+├── gradle
+├── grails-app
+├── src
+├── .gitignore.txt
+├── build.gradle
+├── gradle
+├── gradlew
+├── gradlew.bat
+├── grailsw
+├── grailsw.bat
+└── grails-wrapper
+```
+
+In `gails-app`  has  the following directories
+
+```bash
+├── assets
+├── conf
+├── Controllers
+├── domain
+├── i18n
+├── init
+├── services
+├── taglib
+├── utils
+└── views
+```
+
+### assets folder
+
+This is where all static folders are kept like javascript, Images, CSS and other static multimedia content
+
+### conf folder
+
+This is where configuration files reside this includes resources.groovy, application.yml,logback.groovy, and any other configuration from plugins
+
+### controller folder
+
+The folder contains Controllers, default UrlMapping.groovy which control requests.
+
+### domain folder
+
+Where Groovy classes are kept that are similar to database tables.
+
+### i18n folder
+
+The folder has resources for translation.
+
+### init folder
+
+The folder contains files related to when you launch the web application.
+
+### services folder
+
+All business logic are usually put under the service folder
+
+### taglib folder
+
+This folder contains tags and developers can be able to add their own tags. Tags help in compressing GSP code since helps in the re-use of code
+
+### utils folder
+
+in this folder, we can put utility Groovy classes.
+
+### views folder
+
+it contains GSP files, where HTML code is kept
+
+
+### GRAILS MVC
+
+MVC design patterns help in partitioning responsibilities in the application to simplify the architecture. Model classes represent domain objects in the system. Controller classes control the flow of your application. Service classes control the business logic in the application. Views artifacts are to present information in a way desired.
+
+### Adding more Classes in the `Firstapp` project
+
+### Creating Domain Class
+
+The below command adds an empty Domain Class
+
+```bash
+
+grails create-domain-class Student
+```
+
+```java
+package firstapp
+
+class Student {
+
+    static constraints = {
+    }
+}
+
+```
+
+We can add attributes
+
+```java
+package firstapp
+
+class Student {
+    String name
+    String gender
+    Date dateOfBirth
+    Integer position
+        static constraints = {
+    }
+}
+
+```
+
+### Creating Controller Class
+
+The below command adds an empty Controller Class
```

Comment:
Change "The below command ..." to "The command below ... "

Reply:
sorted